### ISSUE TITLE
"libicui18n.so.67: cannot open shared object file" in Docker

### ISSUE THREAD

**Issue by Half-Shot** [matrix-org/synapse#16509]
Building a Docker image based on v1.94.0 yields this error, so I am wondering if there is a missing dependency?

```
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/usr/local/lib/python3.11/site-packages/synapse/app/homeserver.py", line 37, in <module>
    from synapse.app import _base
  File "/usr/local/lib/python3.11/site-packages/synapse/app/_base.py", line 66, in <module>
    from synapse.handlers.auth import load_legacy_password_auth_providers
  File "/usr/local/lib/python3.11/site-packages/synapse/handlers/auth.py", line 56, in <module>
    from synapse.api.ratelimiting import Ratelimiter
  File "/usr/local/lib/python3.11/site-packages/synapse/api/ratelimiting.py", line 21, in <module>
    from synapse.storage.databases.main import DataStore
  File "/usr/local/lib/python3.11/site-packages/synapse/storage/__init__.py", line 34, in <module>
    from synapse.storage.databases import Databases
  File "/usr/local/lib/python3.11/site-packages/synapse/storage/databases/__init__.py", line 20, in <module>
    from synapse.storage.databases.main.events import PersistEventsStore
  File "/usr/local/lib/python3.11/site-packages/synapse/storage/databases/main/__init__.py", line 76, in <module>
    from .user_directory import UserDirectoryStore
  File "/usr/local/lib/python3.11/site-packages/synapse/storage/databases/main/user_directory.py", line 35, in <module>
    import icu
  File "/usr/local/lib/python3.11/site-packages/icu/__init__.py", line 37, in <module>
    from ._icu_ import *
ImportError: libicui18n.so.67: cannot open shared object file: No such file or directory
```

**Comment by clokep** [comment]
See the conversation at https://matrix.to/#/!vcyiEtMVHIhWXcJAfl:sw1v.org/$B-h4vNy2P1p_-GHpoXJSq9NG5BCLgIbIO81aGdbCqic?via=matrix.org&via=element.io&via=envs.net

tl;dr from @reivilibre is:

> had this a couple weeks ago ish, clear your docker build cache **including build cache mounts**

**Comment by Half-Shot** [comment]
Ah, cool. Thanks!

**Comment by Half-Shot** [comment]
This worked, thanks!
